modules/schema-util/src/main/scala/edu/gemini/schema/util/SourceResolver.scala

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package edu.gemini.schema.util
+
+import scala.io.Source
+import scala.tools.nsc.io.Path
+import cats.effect.Resource
+import cats.effect.Sync
+import cats.effect.syntax.all.*
+import cats.syntax.all.*
+import cats.ApplicativeThrow
+
+// Trait used to find the schema file
+trait SourceResolver[F[_]] {
+  def resolve(name: Path): Resource[F, Source]
+}
+
+object SourceResolver {
+
+  def fromResource[F[_]: Sync](classLoader: ClassLoader): SourceResolver[F] = (name: Path) =>
+    Resource.make(Sync[F].delay(Source.fromResource(name.toString, classLoader)))(x =>
```

Comment:
Could we use `fs2` instead of `Source`, I think you can get a resource as a `InputStream` and pass it to fs2, or `fs2.io` for files